azuredevops/internal/service/serviceendpoint/commons.go

```diff
@@ -381,3 +381,130 @@ func getServiceEndpoint(client *client.AggregatedClient, serviceEndpointID *uuid
 		return nil, opState.Failed, fmt.Errorf(errMsgServiceCreate, serviceEndpointID, *projectID, serviceEndpoint.OperationStatus)
 	}
 }
+
+func dataSourceGenBaseServiceEndpointResource(dataSourceReadFunc schema.ReadFunc) *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceReadFunc,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"service_endpoint_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{"service_endpoint_name", "id"},
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"id": {
```

Comment:
can we rename the `id` to  to `serviceendpoint_id`,   Terraform resource have a default `id` property  same name may result in unexpected behavior.